addons/binding/org.openhab.binding.vitotronic/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,478 @@
+            <channel id="middle_temp" typeId="middle_temp"/>
+            <channel id="bottom_temp" typeId="bottom_temp"/>
+            <channel id="circuitpump" typeId="circuitpump"/>
+        </channels>
+    </thing-type>
+    
+    <thing-type id="circuit">
+    	<supported-bridge-type-refs>
+			<bridge-type-ref id="bridge" />
+		</supported-bridge-type-refs>
+		
+        <label>Circuit</label>
+        <description>Circuit</description>
+
+        <channels>
+            <channel id="flowtemperature" typeId="flow_temp"/>
+            <channel id="pump" typeId="pump"/>
+            <channel id="operationmode" typeId="operationmode"/>
+            <channel id="savemode" typeId="savemode"/>
+            <channel id="partymode" typeId="partymode"/>
+            <channel id="party_temp" typeId="party_temp"/>
+            <channel id="room_temp" typeId="room_temp"/>
+            <channel id="save_temp" typeId="save_temp"/>  
+            <channel id="gradient" typeId="gradient"/>
+            <channel id="niveau" typeId="niveau"/>
+        </channels>
+    </thing-type>
+    
+    <thing-type id="solar">
+    	<supported-bridge-type-refs>
+			<bridge-type-ref id="bridge" />
+		</supported-bridge-type-refs>
+		
+        <label>Circuit</label>
+        <description>Circuit</description>
+
+        <channels>
+            <channel id="collector_temp" typeId="collector_temp"/>
+            <channel id="storagetank_temp" typeId="storagetank_temp"/>
+            <channel id="bufferload" typeId="bufferload"/>
+            <channel id="loadsuppression" typeId="loadsuppression"/>
+            <channel id="ontime" typeId="ontime"/>
+            <channel id="producedheat" typeId="producedheat"/>
+        </channels>
+    </thing-type>
+    
+    <!-- Things for single use -->
+    
+    <thing-type id="temperaturesensor">
+    	<supported-bridge-type-refs>
+			<bridge-type-ref id="bridge" />
+		</supported-bridge-type-refs>
+        <label>TemperatureSensor</label>
+        <description>Single temperature sensor</description>
+        <channels>
+            <channel id="value" typeId="temperature"/>
+        </channels>
+    </thing-type>
+    
+    <thing-type id="pump">
+    	<supported-bridge-type-refs>
+			<bridge-type-ref id="bridge" />
+		</supported-bridge-type-refs>	
+        <label>Pump</label>
+        <description>Single pump</description>
+        <channels>
+            <channel id="value" typeId="pump"/>
+        </channels>
+    </thing-type>
+    
+
+    <thing-type id="valve">
+    	<supported-bridge-type-refs>
+			<bridge-type-ref id="bridge" />
+		</supported-bridge-type-refs>	
+        <label>Valve</label>
+        <description>Single valve</description>
+        <channels>
+            <channel id="valve" typeId="valve"/>
+        </channels>
+    </thing-type>
+    
+
+    <!-- Channel Type -->
+    
+    <!--  Channel Types for heating -->
+    
+    <channel-type id="systemtime">

```

Comment:
have you considered marking a few channels as "advanced"? This will allow the UI to show the most important functionality at first sight (and hide the rest behind a "more" button).